src/RBF/rbf.cpp

```diff
@@ -1181,7 +1381,105 @@ double RBF::calcDist(int i, int j)
  */
 double RBF::calcDist(const std::array<double,3>& point, int j)
 {
-    return norm2(point-m_node[j]);
+    return norm2(point - m_node[j]);
+}
+
+/*!
+ * Initialize activation of monomials terms of linear polynomial part 
+ * If all the nodes are aligned on a plane normal to a cartesian coordinate 
+ * the related coordinate if disable in the polynomial function (no dependency 
+ * can be found).
+ */
+void RBF::initializePolynomialActiveBasis()
+{
+    std::vector<int> activeSet(RBFKernel::getActiveSet());
+
+    // Initialize active terms for an only constant linear polynomial
+    m_polyActiveBasis.clear();
+    m_polyActiveBasis.insert(0);
+
+    // Initialize reference coordinates with the first node
+    std::array<double, 3> coord(m_node[0]);
+
+    // Check if at least one node has one of the coordinates
+    // different from the reference ones and enable the related
+    // polynomial term
+    std::set<int> coordinatesToCheck({0, 1, 2});
+    double tol = 1.0e-12;
+    for (const auto &i : activeSet) {
+        std::array<double, 3> point = m_node[i];
+        for (auto it = coordinatesToCheck.begin(); it != coordinatesToCheck.end();) {
```

Comment:
It seems a quite convoluted way to do a loop over 0, 1, and 2. Maybe something like the following may be easy to understand:
```
for (int d = 0; d < 2; ++d) {
    if (!utils::DoubleFloatingEqual()(coord[d], point[d], tol, tol)) {
        m_polyActiveBasis.insert(d+1);
    }
}
```